cpp/src/arrow/python/helpers.cc

```diff
@@ -254,14 +255,45 @@ bool PyFloat_IsNaN(PyObject* obj) {
   return PyFloat_Check(obj) && std::isnan(PyFloat_AsDouble(obj));
 }
 
+namespace {
+
+static std::once_flag pandas_static_initialized;
+static OwnedRef pandas_NaTType;
```

Comment:
@pitrou this seemed worrisome to me so could use your advice about how to import this symbol once and then have it persist. Another option is that I could immediately decref the object returned by `PyObject_Type` and put a `PyObject*` here with the near certainty that this object will remain alive until the end of the Python session